### PR TITLE
Consolidate ParseCallbacks and Builder to common

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Adds an `SGXParseCallbacks` struct that can be used in a `Builder`.
Adds a function, `sgx_builder()`  to provide a pre-configured `Builder`

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #41** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/41" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/41?utm_source=stack-comment).

**Review by NotGyro — APPROVED** [review]
This is a good consolidation/cleanup, the resulting code is easier to follow. Looks good to me!

File: core/build/src/lib.rs
Comment (samdealy) [inline core/build/src/lib.rs:48]: The codebase normally uses `Sgx...` rather than `SGX...`. Unless there's a reason I'm missing, could you rename this struct to `SgxParseCallbacks`?

File: core/build/src/lib.rs
Comment (samdealy) [inline core/build/src/lib.rs:15]: Nit: Perhaps rename this function to something more descriptive / uses a verb? I'm not sure what it does by just looking at it. Would `add_sgx_item` be an acceptable name?

File: core/ffi/types/build.rs
Comment (samdealy) [inline core/ffi/types/build.rs:16]: What's the reasoning behind adding an explicit `allowlist_type` for each type instead of keeping the for loop? The for loop makes it slightly easier to add a new type becasue you don't need to type out `allowlist_type` again, but I guess this way is slightly more readable?

**Review by samdealy — CHANGES_REQUESTED** [review]

File: core/ffi/types/build.rs
Comment (nick-mobilecoin) [reply core/ffi/types/build.rs:16]: I'm fine reverting, not really invested one way or the other.
 did this as I chewed on this comment, https://github.com/mobilecoinfoundation/sgx/pull/9#discussion_r927103062

I think my thought was to mimic the other crates that are being more explicit about the types, but perhaps they would be better served having a static slice and we make a common utility to add a slice of allowed types

File: core/build/src/lib.rs
Comment (nick-mobilecoin) [reply core/build/src/lib.rs]: Good call out, fixed!

File: core/build/src/lib.rs
Comment (nick-mobilecoin) [reply core/build/src/lib.rs]: I went with `sgx_normalize_item_name()`.  I think that better describes what's going on.

Comment (nick-mobilecoin) [reply core/build/src/lib.rs]: all though the "sgx" prefix may be my C namespace mentality coming in, so feel free to suggest a better name there

File: core/ffi/types/build.rs
Comment (samdealy) [reply core/ffi/types/build.rs]: SGTM.

**Review by samdealy — APPROVED** [review]

File: core/ffi/types/build.rs
Comment (nick-mobilecoin) [reply core/ffi/types/build.rs]: Thinking about it more, I agree the const slice is nicer, I also sorted them, should have done that earlier sorry, but now I know clion's vim bindings support `sort`

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/41?utm_source=stack-comment).


Job Reference: awrIFR4yxeX1nu0a41G3

**Pull request merged**
